Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp

```diff
@@ -5845,25 +5866,62 @@ webkit_web_view_get_default_content_security_policy(WebKitWebView* webView)
     return webView->priv->defaultContentSecurityPolicy.data();
 }
 
+void webkitWebViewSetHasImmersiveSession(WebKitWebView* webView, bool hasImmersiveSession)
+{
+    if (webView->priv->hasImmersiveSession == hasImmersiveSession)
+        return;
+
+    webView->priv->hasImmersiveSession = hasImmersiveSession;
+    g_object_notify_by_pspec(G_OBJECT(webView), sObjProperties[PROP_HAS_IMMERSIVE_SESSION]);
+}
+
+/**
+ * webkit_web_view_get_has_immersive_session:
```

Comment:
Conversely, here we can indicate the property fetched by this getter:

 ```suggestion
 * webkit_web_view_get_has_immersive_session: (get-property has-immersive-session):
```